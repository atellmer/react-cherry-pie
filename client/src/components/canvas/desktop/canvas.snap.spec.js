import React from 'react';
import renderer from 'react-test-renderer';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';

import { windowMatchMedia } from '../../../../test/__mocks__/matchMediaMock';
import muiTheme from '../../../config/theme';
import TmCanvasDesktop from './index';


jest.mock('react-dom');
jest.mock('react-custom-scrollbars', () => {
  return {
    Scrollbars: 'Scrollbars'
  };
});
jest.mock('react-textarea-autosize', () => 'TextareaAutosize');

beforeEach(() => {
  window.matchMedia = windowMatchMedia;
});

test('Jest: TmCanvasDesktop (Snapshot)', () => {
  const props = {
    params: {},
    messagePanelHeight: 0,
    changeMessagePanelHeight: () => {}
  };

  const component = renderer.create(
    <MuiThemeProvider muiTheme={muiTheme}>
      <TmCanvasDesktop {...props}/>
    </MuiThemeProvider>
  );

  let tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});
