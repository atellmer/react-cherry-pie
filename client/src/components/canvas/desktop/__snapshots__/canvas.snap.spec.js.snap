exports[`test Jest: TmCanvasDesktop (Snapshot) 1`] = `
<div
  className={undefined}>
  <div
    className={undefined}
    style={
      Object {
        "paddingBottom": "",
      }
    }>
    <Scrollbars
      autoHide={true}
      autoHideDuration={200}
      autoHideTimeout={1000}
      className={undefined}>
      <div>
        Route: undefined
      </div>
    </Scrollbars>
  </div>
  <div
    className={undefined}>
    <form
      onSubmit={[Function]}>
      <div
        className="Flex"
        style={
          Object {
            "WebkitAlignItems": "center",
            "alignItems": "center",
            "boxSizing": "border-box",
            "display": "flex",
            "msAlignItems": "center",
          }
        }>
        <div
          className="Box"
          style={
            Object {
              "boxSizing": "border-box",
            }
          }>
          <button
            disabled={false}
            href={undefined}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseOut={[Function]}
            onTouchStart={[Function]}
            onTouchTap={[Function]}
            style={
              Object {
                "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
                "background": "none",
                "border": 10,
                "boxSizing": "border-box",
                "cursor": "pointer",
                "display": "inline-block",
                "fontFamily": "Roboto, sans-serif",
                "fontSize": 0,
                "fontWeight": "inherit",
                "height": 48,
                "margin": 0,
                "muiPrepared": true,
                "outline": "none",
                "overflow": "visible",
                "padding": 12,
                "position": "relative",
                "textDecoration": "none",
                "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                "verticalAlign": null,
                "width": 48,
              }
            }
            tabIndex={0}
            type="button">
            <div
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchStart={[Function]}>
              <svg
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                style={
                  Object {
                    "color": "rgba(0, 0, 0, 0.87)",
                    "display": "inline-block",
                    "fill": "#7f7f7f",
                    "height": 24,
                    "muiPrepared": true,
                    "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                    "userSelect": "none",
                    "width": 24,
                  }
                }
                viewBox="0 0 24 24">
                <circle
                  cx="15.5"
                  cy="9.5"
                  r="1.5" />
                <circle
                  cx="8.5"
                  cy="9.5"
                  r="1.5" />
                <path
                  d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8zm0-4c-1.48 0-2.75-.81-3.45-2H6.88c.8 2.05 2.79 3.5 5.12 3.5s4.32-1.45 5.12-3.5h-1.67c-.7 1.19-1.97 2-3.45 2z" />
              </svg>
            </div>
          </button>
        </div>
        <div
          className="Box"
          style={
            Object {
              "WebkitFlex": "1 1 auto",
              "boxSizing": "border-box",
              "flex": "1 1 auto",
              "msFlex": "1 1 auto",
            }
          }>
          <TextareaAutosize
            className={undefined}
            maxRows={10}
            minRows={1}
            onChange={[Function]}
            onHeightChange={[Function]}
            placeholder="Введите ваше сообщение..."
            value="" />
        </div>
        <div
          className="Box"
          style={
            Object {
              "boxSizing": "border-box",
            }
          }>
          <button
            disabled={false}
            href={undefined}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseOut={[Function]}
            onTouchStart={[Function]}
            onTouchTap={[Function]}
            style={
              Object {
                "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
                "background": "none",
                "border": 10,
                "boxSizing": "border-box",
                "cursor": "pointer",
                "display": "inline-block",
                "fontFamily": "Roboto, sans-serif",
                "fontSize": 0,
                "fontWeight": "inherit",
                "height": 48,
                "margin": 0,
                "muiPrepared": true,
                "outline": "none",
                "overflow": "visible",
                "padding": 12,
                "position": "relative",
                "textDecoration": "none",
                "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                "verticalAlign": null,
                "width": 48,
              }
            }
            tabIndex={0}
            type="submit">
            <div
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchStart={[Function]}>
              <svg
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                style={
                  Object {
                    "color": "rgba(0, 0, 0, 0.87)",
                    "display": "inline-block",
                    "fill": "#9a85cd",
                    "height": 24,
                    "muiPrepared": true,
                    "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                    "userSelect": "none",
                    "width": 24,
                  }
                }
                viewBox="0 0 24 24">
                <path
                  d="M2.01 21L23 12 2.01 3 2 10l15 2-15 2z" />
              </svg>
            </div>
          </button>
        </div>
      </div>
    </form>
  </div>
</div>
`;
